js/server/modules/@arangodb/foxx/manifest.js

```diff
@@ -371,27 +377,15 @@ function validateManifestFile (filename, mount, complainAboutVersionMismatches)
   } catch (e) {
     throw Object.assign(
       new ArangoError({
-        errorNum: errors.ERROR_MALFORMED_MANIFEST_FILE.code,
-        errorMessage: dd`
-          ${errors.ERROR_MALFORMED_MANIFEST_FILE.message}
-          File: ${filename}
-        `
-      }), {cause: e}
-    );
-  }
-  try {
-    mf = checkManifest(mf, mount, complainAboutVersionMismatches);
-  } catch (e) {
-    throw Object.assign(
-      new ArangoError({
-        errorNum: errors.ERROR_INVALID_SERVICE_MANIFEST.code,
+        errorNum: ERROR_MALFORMED_MANIFEST_FILE.code,
         errorMessage: dd`
-          ${errors.ERROR_INVALID_SERVICE_MANIFEST.message}
+          ${ERROR_MALFORMED_MANIFEST_FILE.message}
           File: ${filename}
         `
       }), {cause: e}
     );
   }
+  mf = checkManifest(mf, mount, complainAboutVersionMismatches);
```

Comment:
The only errors `checkManifest` should produce normally are manifest validation related, so it doesn't make sense to swallow those and hide them behind an `INVALID_SERVICE_MANIFEST` error. Exposing the error directly also exposes the causes, allowing users to actually fix the problems when using tools like foxx-cli.